WebKit/WebKit
# [LBSE] Provide macOS Ventura pixel baseline for LayoutTests/svg

https://bugs.webkit.org/show_bug.cgi?id=247666

Reviewed by NOBODY (OOPS!).

Regenerate the LBSE pixel test baseline using macOS Ventura, replacing the macOS Monterey results that I previously generated and landed.

Tis still helps LBSE development, locally, when comparing on the same machine, therefore I'm adding a new baseline following the same argumentation as last time: this is not intended for running on the bots at some point, but only to make my life easier during LBSE upstreaming.

The following steps were executed:

1. Regenerate pixel test only baseline for macOS Ventura (only expected.png files in platform/mac-ventura-wk2-pixel)

    $ cd /path/to/WebKit
    $ mkdir LayoutTests/platform/mac-ventura-wk2-pixel
    $ run-webkit-tests --additional-platform-directory=$PWD/LayoutTests/platform/mac-ventura-wk2-pixel --pixel-tests --new-baseline svg
    $ rm LayoutTests/platform/mac-ventura-wk2-pixel/svg/**/*expected.txt

2. Verify pixel test baseline using legacy SVG engine (closure check)

    $ cd /path/to/WebKit
    $ run-webkit-tests --additional-platform-directory=$PWD/LayoutTests/platform/mac-ventura-wk2-pixel --pixel-tests svg

    -> This works as expected, modulo flakey tests.

3. Move LBSE baseline from macOS Monterey to macOS Ventura

    $ git mv LayoutTests/platform/mac-monterey-wk2-lbse-text LayoutTests/platform/mac-ventura-wk2-lbse-text

    -> Use the "old" baseline as starting point for the new one, since it already contains all LBSE-specific render tree dumps.

4. Try LBSE against new pixel test baseline (to assure LBSE delivers same visual results as legacy - where it should)

    $ run-webkit-tests --additional-platform-directory=$PWD/LayoutTests/platform/mac-ventura-wk2-lbse-text \
                       --additional-platform-directory=$PWD/LayoutTests/platform/mac-ventura-wk2-pixel \
                       --additional-expectations=$PWD/LayoutTests/platform/mac-ventura-wk2-lbse-text/TestExpectations \
                       --internal-feature=LayerBasedSVGEngineEnabled=true --pixel-tests --tolerance=0 svg

5. Finish new baseline by hand (remove unncessary pixel result overrides, adapt TestExpectations, etc.)

No change in functionality, thus no new tests.

* LayoutTests/platform/mac-monterey-wk2-lbse-text/TestExpectations: Removed.
...
https://github.com/WebKit/WebKit/commit/bd522fd4bff5614de5d01460c1c1070c37ba4669

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95923 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5172 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28966 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105483 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/165801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5273 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33933 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88301 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/101311 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101585 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/3883 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82530 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/30921 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85737 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/87645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/73754 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39664 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19178 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37344 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/20514 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/41815 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Big-Sur-Debug-WK2-Tests-EWS "Waiting to run tests") | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2158 "Failed to update pull request") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/43723 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/39771 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | |
